pkg/parse/parse.go

```diff
@@ -467,61 +488,93 @@ func GetCacheMount(args []string) (specs.Mount, error) {
 			setShared = true
 		case "bind-propagation":
 			if len(kv) == 1 {
-				return newMount, errors.Wrapf(optionArgError, kv[0])
+				return newMount, "", errors.Wrapf(optionArgError, kv[0])
 			}
 			newMount.Options = append(newMount.Options, kv[1])
 		case "id":
 			id = kv[1]
+		case "from":
+			fromImage = kv[1]
 		case "target", "dst", "destination":
 			if len(kv) == 1 {
-				return newMount, errors.Wrapf(optionArgError, kv[0])
+				return newMount, "", errors.Wrapf(optionArgError, kv[0])
 			}
 			if err := parse.ValidateVolumeCtrDir(kv[1]); err != nil {
-				return newMount, err
+				return newMount, "", err
 			}
 			newMount.Destination = kv[1]
 			setDest = true
+		case "src", "source":
+			if len(kv) == 1 {
+				return newMount, "", errors.Wrapf(optionArgError, kv[0])
+			}
+			if err := parse.ValidateVolumeHostDir(kv[1]); err != nil {
+				return newMount, "", err
+			}
+			newMount.Source = kv[1]
```

Comment:
@nalind In case of `type=cache` field `src` is only applicable when `from` is specified. So in the same function below we change the `contextdir` to the mount created by `from` and if `from` is not specified `src` is just ignored.

Reply:
@nalind in the same function we override `src` if `from` is not specified here so it happens before `validateVolumeHostDir()` https://github.com/containers/buildah/pull/3590/files#diff-0c73d3e5eb67eaabc39caa4cbb5dd69eb43ae6d98b1ca8889fc86d56bacefb90R563